Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift

```diff
@@ -37,29 +37,208 @@ import Markdown
 /// Block elements can be nested, for example, an aside note contains one or more paragraphs of text.
 public enum RenderBlockContent: Equatable {
     /// A paragraph of content.
-    case paragraph(inlineContent: [RenderInlineContent])
+    case paragraph(Paragraph)
     /// An aside block.
-    case aside(style: AsideStyle, content: [RenderBlockContent])
+    case aside(Aside)
     /// A block of sample code.
-    case codeListing(syntax: String?, code: [String], metadata: RenderContentMetadata?)
+    case codeListing(CodeListing)
     /// A heading with the given level.
-    case heading(level: Int, text: String, anchor: String?)
+    case heading(Heading)
     /// A list that contains ordered items.
-    case orderedList(items: [ListItem])
+    case orderedList(OrderedList)
     /// A list that contains unordered items.
-    case unorderedList(items: [ListItem])
+    case unorderedList(UnorderedList)
     
     /// A step in a multi-step tutorial.
-    case step(content: [RenderBlockContent], caption: [RenderBlockContent], media: RenderReferenceIdentifier?, code: RenderReferenceIdentifier?, runtimePreview: RenderReferenceIdentifier?)
+    case step(TutorialStep)
     /// A REST endpoint example that includes a request and the expected response.
-    case endpointExample(summary: [RenderBlockContent]?, request: CodeExample, response: CodeExample)
+    case endpointExample(EndpointExample)
     /// An example that contains a sample code block.
-    case dictionaryExample(summary: [RenderBlockContent]?, example: CodeExample)
+    case dictionaryExample(DictionaryExample)
     
     /// A list of terms.
-    case termList(items: [TermListItem])
+    case termList(TermList)
     /// A table that contains a list of row data.
-    case table(header: HeaderType, rows: [TableRow], metadata: RenderContentMetadata?)
+    case table(Table)
+
+    // Warning: If you add a new case to this enum, make sure to handle it in the Codable
+    // conformance at the bottom of this file, and in the `rawIndexableTextContent` method in
+    // RenderBlockContent+TextIndexing.swift!
```

Comment:
Hm I hadn't considered this. I guess ideally we wouldn't want to mark the the `_nonfrozenEnum_useDefaultCase` case as deprecated _internally_ and only extenrally so that we can still switch on it within the library and make sure SwiftDocC itself is covering all of bases.

What do you think of the trade-off here? Should we consider not deprecating it so we can switch over it internally? Then we would move the deprecation message into a documentation comment instead.